core/src/main/java/org/apache/calcite/rel/rules/FilterJoinRule.java

```diff
@@ -53,28 +52,21 @@
    * will be pushed into the ON clause. */
   public static final Predicate TRUE_PREDICATE = (join, joinType, exp) -> true;
 
-  /** Predicate that returns true if the join is not Enumerable convention,
-   * will be replaced by {@link #TRUE_PREDICATE} once enumerable join supports
-   * non-equi join. */
-  // to be removed before 1.22.0
-  private static final Predicate NOT_ENUMERABLE = (join, joinType, exp) ->
-      join.getConvention() != EnumerableConvention.INSTANCE;
-
   /** Rule that pushes predicates from a Filter into the Join below them. */
   public static final FilterJoinRule FILTER_ON_JOIN =
```

Comment:
We can remove the predicate member if it always returns true.

Reply:
+1
I guess the intention here is for backward compatibility, since the constructor of FilterIntoJoinRule is public

Reply:
I don't mind removing it if you think the backward  compatibility is not a concern.